client-ts/Microsoft.AspNetCore.SignalR.Client.TS/IHubProtocol.ts

```diff
@@ -28,8 +28,14 @@ export interface NegotiationMessage {
     readonly protocol: string;
 }
 
+export const enum ProtocolType {
+    Text = 1,
+    Binary
+}
+
 export interface IHubProtocol {
     name(): string;
+    type(): ProtocolType;
```

Comment:
on the other hand you could replace the entire function in JS so meh - can change to a property (same for `name()`)